src/pypromice/process/L2toL3.py

```diff
@@ -315,72 +388,82 @@ def calcHumid(T_0, T_100, T_h, es_0, es_100, eps, p_h, RH_cor_h):
-        | np.isnan(RH_cor) | np.isnan(WS) | np.isnan(z_boom)
-    SHF[HF_nan] = np.nan
-    LHF[HF_nan] = np.nan 
-    return SHF, LHF
-      
-def cleanSpHumid(q_h, T, Tsurf, p, RH_cor):
-    '''Find invalid specific humidity data values and replace with NaNs, 
-    based on air temperature, surface temperature, air pressure, 
-    and corrected relative humidity
+    df_all = pd.Series(dtype=float)  # Initialize an empty Series to gather all smoothed pieces
+    breaks = [None] + breaks + [None]
+    for i in range(len(breaks) - 1):
+        df = df_in.loc[slice(breaks[i], breaks[i+1])].copy()
+               
+        # Drop NaN values and calculate the number of segments based on valid data
+        df_valid = df.dropna()
+        if df_valid.shape[0] > 2:        
+            # Fit linear regression model to the valid data range
+            x = pd.to_numeric(df_valid.index).values.reshape(-1, 1)
+            y = df_valid.values.reshape(-1, 1)
+            
+            model = LinearRegression()
+            model.fit(x, y)
+            
+            # Predict using the model for the entire segment range
+            x_pred = pd.to_numeric(df.index).values.reshape(-1, 1)
+            
+            y_pred = model.predict(x_pred)
+            df =  pd.Series(y_pred.flatten(), index=df.index)
+
+        
+        # Update df_all with predicted values for the current segment
+        df_all = pd.concat([df_all, df])
     
-    Parameters
-    ----------
-    q_h : xarray.DataArray
-        Specific humidity
-    T : xarray.DataArray
-        Air temperature
-    Tsurf : xarray.DataArray
-        Surface temperature
-    p : xarray.DataArray
-        Air pressure
-    RH_cor : xarray.DataArray
-        Relative humidity corrected
+    # Fill internal gaps with linear interpolation
+    df_all = df_all.interpolate(method='linear', limit_area='inside')
     
-    Returns
-    -------
-    q_h : xarray.DataArray
-        Specific humidity corrected'''
-    q_nan = np.isnan(T) | np.isnan(RH_cor) | np.isnan(p) | np.isnan(Tsurf)
-    q_h[q_nan] = np.nan
-    return q_h
- 
+    # Extrapolate for timestamps before the first valid measurement
+    first_valid_6_months = slice(None, df_in.first_valid_index() + pd.to_timedelta('180D'))
+    df_all.loc[first_valid_6_months] = (
+        df_all.loc[first_valid_6_months].interpolate(
+            method='linear', limit_direction='backward', fill_value="extrapolate"
+        )
+    ).values
+    
+    # Extrapolate for timestamps after the last valid measurement
+    last_valid_6_months = slice(df_in.last_valid_index() - pd.to_timedelta('180D'), None)
+    df_all.loc[last_valid_6_months] = (
+        df_all.loc[last_valid_6_months].interpolate(
+            method='linear', limit_direction='forward', fill_value="extrapolate"
+        )
+    ).values
+    
+    # Remove duplicate indices and return values as numpy array
+    df_all = df_all[~df_all.index.duplicated(keep='first')]
```

Comment:
I don't see how this function can cause duplicated indices

Reply:
Because for each iteration/segment we select:
https://github.com/GEUS-Glaciology-and-Climate/pypromice/blob/f66251fd6df1e5899f9e43de5ce417694b7c9a6f/src/pypromice/process/L2toL3.py#L422
which includes the timestamps of both breaks. So at one iteration the end break received an estimated value, and in the next iteration/segment that same break is a start break ad receives another value.

We therefore need to drop duplicates at the end. 
I'm changing to `keep='last'` so that if a time stamp is a break, then it received the value estimated within the segment where it is a start break (and not an end break).